app/services/schools/milestone_sort_service.rb

```diff
@@ -0,0 +1,52 @@
+module Schools
+  class MilestoneSortService
+    def initialize(target, direction)
+      @target = target
+      @direction = direction
+    end
+
+    def execute
+      index = milestones.index(target)
+
+      if (index < 1 && should_move_up) ||
+           (index >= milestones.size - 1 && !should_move_up)
+        return
+      end
+
+      target_2 = milestones[should_move_up ? index - 1 : index + 1]
+
+      Target.transaction do
+        number = target.milestone_number
+        target.update!(milestone_number: target_2.milestone_number)
+        target_2.update!(milestone_number: number)
+      end
```

Comment:
This can be avoided using arrays. Here we have two unnecessary DB calls just to swap two targets. Instead, create an array of milestone target IDs, perform the swap in the array of IDs. And then you can do something similar to: https://github.com/pupilfirst/pupilfirst/blob/9b55c4d1a6912c53788e35bceba71f663ff6fe2c/app/queries/sort_curriculum_resources_mutator.rb#L12C7-L12C7